torchfields/voting.py

```diff
@@ -48,19 +48,40 @@ def vote(self, softmin_temp=1, blur_sigma=1):
         DisplacementField of shape (1, 2, H, W) containing the vector
         voting result
     """
+    field_weights = get_field_weights(self, 
+                                      softmin_temp=softmin_temp,
+                                      blur_sigma=blur_sigma)
+    partition = field_weights.sum(dim=0, keepdim=True)
+    field_weights = field_weights / partition
+    return (self * field_weights.unsqueeze(-3)).sum(dim=0, keepdim=True)
+
+def get_field_weights(field, softmin_temp=1, blur_sigma=1):
+    """Calculate softmin weights for set of fields, indicating which fields
+    should be considered consensus.
+
+    Args:
+        field: DisplacementField of shape (N, 2, H, W)
+        softmin_temp (float): temperature of softmin to use
+        blur_sigma (float): std dev of the Gaussian kernel by which to blur
+            the softmin inputs. Note that the outputs are not blurred.
+            None or 0 means no blurring.
+
+    Returns:
+        torch.Tensor of shape (N, 1, H, W) containing the field weights 
+    """
     from itertools import combinations
-    if self.ndimension() != 4:
+    if field.ndimension() != 4:
```

Comment:
Why change the variable name to `field` here?
These functions are factored out, but in reality, they serve as methods of the [`DisplacementField` class](https://github.com/seung-lab/torchfields/blob/894dc2b639e45cd2f7a7b70a9a7fce960def2d49/torchfields/fields.py#L884-L892).

If this function is intended to be public, then it should also be added there so that it can be called as 
`x.get_field_weights(...)`